daemon/changes.go

```diff
@@ -22,6 +22,9 @@ func (daemon *Daemon) ContainerChanges(name string) ([]archive.Change, error) {
 
 	container.Lock()
 	defer container.Unlock()
+	if container.RWLayer == nil {
```

Comment:
We should probably check if the container is in the dead state first.
This would allow us to give a better error message.

Reply:
Right, I have changed this part (also to not use the returned data in case err was not nil). Please see the updated patch